gramps/gen/db/upgrade.py

```diff
@@ -317,6 +317,8 @@ def gramps_upgrade_17(self):
             n -= 1
 
         if parent_handle is not None:
+            if isinstance(parent_handle, str):
+                parent_handle = parent_handle.encode()
             placeref_list = [(parent_handle.decode("utf-8"), None)]
```

Comment:
This seems only used here; it would be simpler to set `placeref_list` directly instead of encoding and decoding again.
```suggestion
            if isinstance(parent_handle, str):
                placeref_list = [(parent_handle, None)]
            else:
                placeref_list = [(parent_handle.decode("utf-8"), None)]
```